integration/utils/DriverFactory.js

```diff
@@ -0,0 +1,58 @@
+const { Builder } = require('selenium-webdriver');
+const firefox = require('selenium-webdriver/firefox');
+const chrome = require('selenium-webdriver/chrome');
+const config = require('./config');
+
+class DriverFactory {
+  configure(host) {
+    let builder = new Builder();
+    const sauceLabsURL = 'https://ondemand.saucelabs.com/wd/hub';
+
+    switch (host) {
+      case 'sauce-firefox':
+        builder.usingServer(sauceLabsURL);
+        builder.withCapabilities({
+          ...config.firefoxOptions,
```

Comment:
Checking for the order here: If `sauceOptions` have some attributes in common with `firefoxOptions` are we going to override any `firefoxOptions`? OR this is not the case.

Reply:
`sauceOptions` stores Sauce Labs related config. `firefoxOptions` and `chromeOptions` store the browser related config. So this is not the case.

```js
  sauceOptions: {
    browserVersion: process.env.BROWSER_VERSION || 'latest',
    platformName: process.env.PLATFORM_NAME || 'Windows 10',
    'sauce:options': {
      username: process.env.SAUCE_USERNAME,
      accessKey: process.env.SAUCE_ACCESS_KEY,
      tunnelIdentifier: process.env.JOB_ID,
      noSSLBumpDomains: 'all',
      extendedDebugging: true,
    }
  },
```
```js
  firefoxOptions: {
    browserName: 'firefox',
    resolution: '1920x1080',
    'moz:firefoxOptions': {
      args: [
        '-start-debugger-server',
        '9222'
      ],
      prefs: {
        'media.navigator.streams.fake': true,
        'media.navigator.permission.disabled': true,
        'media.peerconnection.video.h264_enabled': true,
        'media.webrtc.hw.h264.enabled': true,
        'media.webrtc.platformencoder': true,
        'devtools.chrome.enabled': true,
        'devtools.debugger.prompt-connection': false,
        'devtools.debugger.remote-enabled': true
      },
    },
  },
  chromeOptions: {
    browserName: 'chrome',
    resolution: '1920x1080',
    'goog:chromeOptions': {
      args: [
        '--use-fake-ui-for-media-stream',
        '--use-fake-device-for-media-stream',
      ],
    },
  },
```

Reply:
Yup I checked the config later in the review. Thanks for explanation.